### DISCUSSION TITLE
IPFS hosting with static `assetPrefix`

### DISCUSSION THREAD

**Discussion by NoahZinsmeister** [vercel/next.js#10745] [Help]
Hi! I'm trying to host a fully statically exported Next.js site on both DNS as well as IPFS. So far, I've achieved this by using a build-time flag, and optionally prefixing a tag `href`s, img tag `src`s, etc., with `'.'` with a custom base tag so that my site can be served on e.g. `https://ipfs.io/ipns/<...>/`. I also have to make assetPrefix `'.'` when building for IPFS. The problem is that once I'm >1 route deep, assetPrefix breaks, as it begins looking for shared `_next` resources relative to the current path, whereas it should actually be looking relative to my custom base tag.

One solution to this may be a dynamic `assetPrefix`, but as far as I can tell this is impossible in statically exported sites. Is there another possible solution that I'm missing?

**Comment by foxundermoon** [comment]
I have a solution.
will be posting a blog later if you need it.

code is here  https://github.com/DnCov/transparent-info/blob/master/.github/workflows/auto_release.yml
